wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -850,6 +855,174 @@ instance BuildableSafeGen Wallet where
 instance Buildable [Wallet] where
     build = bprint listJson
 
+
+
+--------------------------------------------------------------------------------
+-- Utxo statistics
+--------------------------------------------------------------------------------
+
+-- | Utxo statistics for the wallet.
+-- | Histogram is composed of bars that represent the bucket. The bucket is tagged by upper bound of a given bucket.
+-- | The bar value corresponds to the number of stakes
+-- | In the future the bar value could be different things:
+-- | (a) sum of stakes in a bucket
+-- | (b) avg or std of stake in a bucket
+-- | (c) topN buckets
+-- | to name a few
+newtype HistogramBar = HistogramBarCount (Text, Word64) deriving (Show, Eq, Ord, Generic)
+
+instance ToSchema HistogramBar where
+  declareNamedSchema = genericDeclareNamedSchema defaultSchemaOptions
+
+instance Arbitrary HistogramBar where
+  arbitrary =
+      let possibleBuckets = fmap show $ (zipWith (\ten toPower -> ten^toPower :: Word64) (repeat (10::Word64)) [(1::Word64)..16]) ++ [45 * (10^(15::Word64))]
+          possibleBars = zipWith (curry HistogramBarCount) possibleBuckets [0..]
+      in elements possibleBars
+
+deriveSafeBuildable ''HistogramBar
+instance BuildableSafeGen HistogramBar where
+    buildSafeGen _ (HistogramBarCount pair) =
+        bprint build pair
+
+
+data UtxoStatistics = UtxoStatistics
+  { theHistogram :: ![HistogramBar]
+  , theAllStakes :: !Word64
+  } deriving (Show, Generic, Ord)
+
+toMap :: [HistogramBar] -> Map Text Word64
+toMap = Map.fromList . map (\(HistogramBarCount (key, val)) -> (key,val))
+
+instance Eq UtxoStatistics where
+    (UtxoStatistics h s) == (UtxoStatistics h' s') = s == s' && toMap h == toMap h'
+
+instance ToJSON UtxoStatistics where
+    toJSON (UtxoStatistics bars allStakes) =
+        let histogramObject = Object . HMS.fromList . map extractBarKey
+            extractBarKey (HistogramBarCount (bound, stake)) = bound .= stake
+        in object [ "histogram" .= histogramObject bars
+                  , "allStakes" .= allStakes ]
+
+instance FromJSON UtxoStatistics where
+    parseJSON (Object v) =
+        let histogramListM = case HMS.lookup "histogram" v of
+                Nothing   -> empty
+                Just (Object bars) -> do
+                    let constructHistogram (key, Number val) =
+                            case floatingOrInteger val of
+                                Left (_ :: Double)        -> HistogramBarCount ("0", 0 :: Word64)
```

Comment:
What about failing explicitly here? If there's a non-integer value, that's definitely not a valid `UtxoStatistics` being parsed. 
Also, we use `Word64` everywhere whereas what we really want are `Natural` I believe ?